lib/matplotlib/cm.py

```diff
@@ -255,10 +254,18 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                     xx = x
                 else:
                     raise ValueError("third dimension must be 3 or 4")
-                if bytes and xx.dtype != np.uint8:
-                    xx = (xx * 255).astype(np.uint8)
-                if not bytes and xx.dtype == np.uint8:
-                    xx = xx.astype(float) / 255
+                if xx.dtype.kind == 'f':
+                    if xx.max() > 1 or xx.min() < 0:
```

Comment:
Having poked at it, for image I think it's impossible for this case to even happen -- the data is normalized (in any event) before it gets here.  (However, I suspect that's not the case for other places to_rgba is used, like scatter).